hexa/pipelines/models.py

```diff
@@ -404,6 +411,14 @@ def add_output(self, uri: str, output_type: str, name: typing.Optional[str]):
         )
         self.save()
 
+    def stop_if_has_perm(self, principal: User):
```

Comment:
it can simply be `def stop(self, principal: User)` and the check is made inside it. I think we only use the "if_has_perm" for basic CRUD operations

Reply:
Copy that.